### PR TITLE
CDC: My Account-> Consent Management

### PR DESCRIPTION
**Pull request by anjana-bl**

### CODE REVIEW

File: projects/storefrontlib/cms-components/myaccount/consent-management/components/consent-management.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Injectable()
+export class ConsentManagementService {
+  constructor(protected anonymousConsentsConfig: AnonymousConsentsConfig) {}
+  fillRequiredConsents(): string[] {
+    if (this.anonymousConsentsConfig.anonymousConsents) {
+      if (this.anonymousConsentsConfig.anonymousConsents.requiredConsents) {
+        return this.anonymousConsentsConfig.anonymousConsents.requiredConsents;
+      }
+    }
+    return [];
```
Comment (Platonn) [inline projects/storefrontlib/cms-components/myaccount/consent-management/components/consent-management.service.ts]: tip: can be written in 1 line:
```ts
return this.anonymousConsentsConfig?.anonymousConsents?.requiredConsents || [];
```

File: integration-libs/cdc/components/consent-management/cdc-consent-management.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          if (
+            consent?.required === true &&
+            consent.id &&
+            consent.currentConsent?.consentGivenDate
+          )
+            requiredConsents.push(consent.id);
+        }
+      })
+      .unsubscribe();
+    return requiredConsents;
```
Comment (Platonn) [inline integration-libs/cdc/components/consent-management/cdc-consent-management.service.ts]: 1. If possible, don't mix observable and static paradigms.
2. do you really have to subscribe to `this.userConsentService.getConsents()`? Cant you pass the consent templates as the argument of `fillRequiredConsents(consents)` method?

File: projects/storefrontapp/src/app/spartacus/features/cdc/cdc-feature.module.ts [hunk 3]
```diff
@@ -15,12 +16,19 @@ import { CmsConfig, provideConfig } from '@spartacus/core';
       cdc: [
         {
           baseSite: 'electronics-spa',
-          javascriptUrl: 'JS_SDK_URL_PLACEHOLDER',
+
+          javascriptUrl:
```
Comment (SherwinVarghese) [inline projects/storefrontapp/src/app/spartacus/features/cdc/cdc-feature.module.ts:20]: @anjana-bl please update this line to remove API Keys used in development.

File: projects/storefrontapp/src/app/spartacus/features/cdc/cdc-feature.module.ts [hunk 4]
```diff
@@ -15,12 +16,19 @@ import { CmsConfig, provideConfig } from '@spartacus/core';
...
+
           sessionExpiration: 3600,
         },
+
         {
           baseSite: 'powertools-spa',
-          javascriptUrl: 'JS_SDK_URL_PLACEHOLDER',
+
+          javascriptUrl:
+            'https://cdns.eu1.gigya.com/JS/gigya.js?apikey=3__pAj9UsaNXJAaDi-d8xvhzNBvGXDYx0GlTg1R9YQWVgTIQbdBZzR98_y-nFZWUNl',
```
Comment (SherwinVarghese) [inline projects/storefrontapp/src/app/spartacus/features/cdc/cdc-feature.module.ts:30]: @anjana-bl please update this line to remove API Keys used in development.

File: integration-libs/cdc/components/consent-management/converters/cdc-site-consent.normalizer.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { Converter, ConsentTemplate, LanguageService } from '@spartacus/core';
+import {
+  CdcSiteConsentTemplate,
+  siteConsentDetailTemplate,
+} from 'integration-libs/cdc/core';
```
Comment (SherwinVarghese) [inline integration-libs/cdc/components/consent-management/converters/cdc-site-consent.normalizer.ts:12]: you may use relative imports here as pointed out by Github actions.

File: .vscode/launch.json [hunk 6]
```diff
@@ -1,63 +1,22 @@
 {
-  // Use IntelliSense to learn about possible attributes.
```
Comment (SherwinVarghese) [inline .vscode/launch.json:2]: Please revert changes and avoid checking in this file

File: integration-libs/cdc/root/service/cdc-js.service.ts [hunk 7]
```diff
@@ -610,6 +614,48 @@ export class CdcJsService implements OnDestroy {
...
+    let siteApiKey: string | null = queryParams.get('apikey');
+
+    return this.invokeAPI('accounts.getSiteConsentDetails', {
+      apiKey: siteApiKey,
+    });
+  }
+
+  protected currentUID: string | undefined;
+
+  getUserConsentPreferences(
```
Comment (SherwinVarghese) [inline integration-libs/cdc/root/service/cdc-js.service.ts:632]: This `getUserConsentPreferences` can be avoided. In https://github.com/SAP/spartacus/blob/3ff03d7d421fe8756bd8d82e4d8d85037a07d96e/integration-libs/cdc/root/service/cdc-js.service.ts#L124 `preferences` can be included as well.

So the preferences can be obtained on login itself.

**Comment by anjana-bl** [comment]
closing the PR as still some development is pending

**Pull request closed without merging**